src/app/shared/components/template/services/instance/template-row.service.ts

```diff
@@ -327,30 +333,32 @@ export class TemplateRowService extends TemplateInstanceService {
    * @param items - list of items to iterate over
    */
   private generateLoopItemRows(row: FlowTypes.TemplateRow, items: any[]) {
-    let item_rows = items.map((item, index) => {
+    const loopItemRows: FlowTypes.TemplateRow[] = [];
+    for (const [index, item] of Object.entries(items)) {
       item._index = index;
       const evalContext = { itemContext: item };
-      const itemRow: FlowTypes.TemplateRow = {
-        type: "group",
-        rows: row.rows.map((r) => this.setRecursiveRowEvalContext(r, evalContext)),
-      } as any;
-      return itemRow;
-    });
-    return item_rows;
+      for (const r of row.rows) {
+        const itemRow = this.setRecursiveRowEvalContext(r, evalContext);
+        loopItemRows.push(itemRow);
+      }
+    }
+    return loopItemRows;
   }
   /** Update the evaluation context of a row and recursively any nested rows */
   private setRecursiveRowEvalContext(
     row: FlowTypes.TemplateRow,
     evalContext: FlowTypes.TemplateRow["_evalContext"]
   ) {
-    const { rows, ...rest } = row;
```

Comment:
memory allocation fix, when it comes to operating on deeply nested objects issues can arise in edge cases where re-assigning and referring to deeply nested objects with the same may sometimes keep the wrong item in memory. Hacky fix is to convert to string and back again as this always creates a new memory allocation.